tests/spec/core/dfn-spec.js

```diff
@@ -655,5 +655,29 @@ describe("Core — Definitions", () => {
       expect(errors[0].message).toContain("Declares both");
       expect(errors[1].message).toContain("but also has a");
     });
+
+    it("assigns data-defines on well-known patterns", async () => {
+      const body = `
+        <section>
+          <h2>Definition and its <dfn>description</dfn></h2>
+          <p id="desc3" class="definition">A description explains what a term is.</p>
```

Comment:
I think this is too magical. We should only rely on sibling order where HTML itself relies on order (i.e., only for `dt+dd`). Other should require explicit association through nesting (and/or `data-defines`).

Reply:
a heading with a `<dfn>` followed by a nearby definition of the term is pretty frequent pattern, but I certainly wouldn't object to leave that to a future improvement. @tidoust?

Reply:
Totally fine to start without it!